### DISCUSSION TITLE
Controlling administrator windows

### DISCUSSION THREAD

**Discussion by vertigo220** [WindowTop/WindowTop-App#403] [General]
Regarding this statement from the releases page:

> Currently, there is a known issue where it does not function on windows with higher privileges (elevated/admin windows) unless WindowTop is also running as Admin. I have spent some time trying to find a workaround, but unfortunately, no solution has been found yet.

you may find this useful:

https://learn.microsoft.com/en-us/windows/powertoys/administrator

I don't think a workaround is possible. I've struggled with this issue myself in one of my own projects, and even Microsoft's own app is pretty clear that it can't be done. So probably not worth the time trying to figure it out.

**Comment by gileli121** [comment]
Hello @vertigo220,  

Thanks for the feedback.  

Actually, a workaround is possible and already implemented, though I can't find a working workaround for this specific hotkey at the moment. 

The existing workaround allows WindowTop to handle elevated/admin windows seamlessly. When you attempt to use one of its features on an elevated/admin window, WindowTop prompts a confirmation request via the OS, asking for permission to run as admin. Behind the scenes, WindowTop launches a secondary helper process with admin privileges. This process communicates with the main application whenever an elevated/admin window is detected.  

On the first attempt to enable a feature like Always On Top, the secondary process isn't running yet. At this point, the OS prompts you with a confirmation request (via UAC). Once approved, the secondary process starts, and subsequent commands for elevated/admin windows are sent directly to this process with admin permissions.  

You can read more about it here: https://github.com/WindowTop/WindowTop-App/issues/241#issuecomment-1172956275  

This is how it works. PowerToys does not implement a similar mechanism. In fact, WindowTop is more sophisticated than PowerToys in many ways. Just because PowerToys carries the "Microsoft" name doesn't make it inherently better. Their developers don't have access to any exclusive internal OS details or APIs.

**Reply by vertigo220** [reply]
> Just because PowerToys carries the "Microsoft" name doesn't make it inherently better. Their developers don't have access to any exclusive internal OS details or APIs.

Believe me, I realize that. If anything, IMO it makes it worse, as I find MS to be _extremely_ incompetent and have found that _every single_ one of their products, hardware and software, are full of problems. I did, however, figure that if anyone would know how to get around this issue, it would be them, but I guess I should have known better.
